### PR TITLE
Add React Native sdk.

### PR DESCRIPTION
**Pull request by ankit-adapt**
# Summary

This PR introduces the `@outspeed/react-native` package, which provides WebRTC functionality for streaming audio and video in React Native applications. It leverages some utility functions from `@outspeed/core` (such as the `SDP` class and `fetchWithRetry` function).

# Changes

1. **New Package**: Added a new package under `src/packages` called `@outspeed/react-native`.
2. **WebRTC Support**:
   - Add the React Native SDK to connect to the WebRTC backend, enabling both audio and video streaming.
   - Supports receiving audio and video streams from the backend.
3. **New Components**:
   - **`RealtimeConnection`**: Manages the WebRTC connection lifecycle.
   - **`RealtimeNegotiator`**: Handles connection negotiation logic.
   - **`RealtimeMediaManager`**: Manages media permissions.
   - **`useWebRTC` hook**: Simplifies peer connection management by wrapping `RealtimeConnection` for React components.

# Tests

- **Android**:
  - Tested streaming audio and video successfully in an Android emulator.
  - Verified data channel functionality.
  
- **iOS**:
  - Tested streaming audio successfully in an iOS emulator.
  - Unable to test video streaming in the iOS emulator (due to the limitation of not being able to share the camera), but expect it to work as the underlying logic for audio and video is similar.
  - Data channel functionality was tested and is working as expected on both platforms.

#### Test on iOS

To test this on iOS, follow these steps:

1. **Install Xcode**: Download and install it from [Xcode on the App Store](https://apps.apple.com/us/app/xcode/id497799835).
2. **Install Xcode Command Line Tools**: Open Xcode, go to **Settings** (or press `cmd ⌘ + ,`), navigate to the **Locations** tab, and install the tools by selecting the latest version from the **Command Line Tools** dropdown.
3. **Use the Simulator**: You can refer to this [YouTube tutorial](https://www.youtube.com/watch?v=5DbMomHr5ys) for help with the simulator.
4. **CocoaPods Issues**: If you encounter any issues while downloading CocoaPods on your Mac, follow this [tutorial](https://www.youtube.com/watch?v=uZD2EQbBqPg).

Run the following command to run the [reat-native-playground](https://github.com/outspeed-ai/react-native-playground) in iOS:

```bash
npx expo prebuild
```

```bash
npx expo run:ios
```

### **Note**

None.

### CODE REVIEW

File: packages/react/package.json [hunk 1]
```diff
@@ -36,7 +36,8 @@
   },
   "peerDependencies": {
     "react": "^18.2.0",
-    "react-dom": "^18.2.0"
+    "react-dom": "^18.2.0",
+    "@outspeed/core": "workspace:^"
```
Comment (janak2) [inline packages/react/package.json]: why did we move it to peerdependency?

**Comment by janak2** [comment]
Let's keep the naming consistent. Rename react-native-sdk to react-native

File: packages/react/package.json (hunk 1 above)
Comment (ankit-adapt) [reply packages/react/package.json]: Yes, we don't need to add this in peerDependencies. I will revert this change.

**Comment by ankit-adapt** [comment]
> Let's keep the naming consistent. Rename react-native-sdk to react-native

Makes sense, I will rename it to `react-native`

**Pull request merged**